examples/vision/siamesenetwork.py

```diff
@@ -0,0 +1,417 @@
+"""
+Title: Siamese Network with Triplet Loss
+Author: [Hazem Essam](https://twitter.com/hazemessamm) and [Santiago L. Valdarrama](https://twitter.com/svpino)
+Date created: 2021/03/13
+Last modified: 2021/03/22
+Description: Siamese network with tf.data and custom training loop.
+"""
+
+
+"""
+TODO: Still need to work on this.
+
+# Introduction
+
+[Siamese Network](https://en.wikipedia.org/wiki/Siamese_neural_network) is used to solve
+many problems like detecting question duplicates, face recognition by comparing the
+similarity of the inputs by comparing their feature vectors.
```

Comment:
Suggestion: use active voice vs passive + "speak" to the user

```suggestion
You can use a [Siamese Network](https://en.wikipedia.org/wiki/Siamese_neural_network) 
to solve various problems in machine learning, such as detecting question duplicates, face recognition through a comparison of the similarity of the inputs by comparing their feature
vectors.
```

(trying to minimize the "comparing... comparing" repetition)